dot/rpc/modules/author.go

```diff
@@ -94,27 +116,66 @@ func NewAuthorModule(logger log.Logger, coreAPI CoreAPI, txStateAPI TransactionS
 	}
 }
 
-// InsertKey inserts a key into the keystore
-func (am *AuthorModule) InsertKey(r *http.Request, req *KeyInsertRequest, res *KeyInsertResponse) error {
-	keyReq := *req
+// HasSessionKeys checks if the keystore has private keys for the given session public keys.
+func (am *AuthorModule) HasSessionKeys(r *http.Request, req *HasSessionKeyRequest, res *HasSessionKeyResponse) error {
+	pubKeysBytes, err := common.HexToBytes(req.PublicKeys)
+	if err != nil {
+		return err
+	}
+
+	pkeys, err := scale.Marshal(pubKeysBytes)
+	if err != nil {
+		return err
+	}
 
-	pkDec, err := common.HexToBytes(keyReq[1])
+	data, err := am.coreAPI.DecodeSessionKeys(pkeys)
 	if err != nil {
+		*res = false
 		return err
 	}
 
-	privateKey, err := keystore.DecodePrivateKey(pkDec, keystore.DetermineKeyType(keyReq[0]))
+	var decodedKeys *[]decodedKey
+	err = scale.Unmarshal(data, &decodedKeys)
+	if err != nil {
+		return err
+	}
+
+	if decodedKeys == nil {
+		*res = false
+		return errors.New("could not decode DecodeSessionKeys response")
```

Comment:
is this what it means? I would assume that the return was `nil`.

Reply:
sorry, the error message is not accurate. Ok, I will remove and return just `nil`

Reply:
I've added the condition `len(*decodedKeys) < 1)` to prevent an empty slice of `keys` being valid